Azure/azure-container-networking
# ci: [NPM] scale test pipeline using KWOK

**Reason for Change**:
Automate NPM scale tests.

**Issue Fixed**:

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)

**Notes**:
### Test Steps
#### 1. Scale Up Large (~1K fake Pods, 30 real Pods)
- Create Fake + Real Pods.
- Label them.
- Add NetPols.

#### 2. Test NPM Bootup Latency and Connectivity
Must achieve initial connectivity within 15 minutes, and apply a new NetworkPolicy within 30 seconds.

- Restart NPM to see performance improvements from #1900.
- Check connectivity. Fail if a timeout is exceeded.
  - Create two pinger Pods.
  - Check initial connectivity. If unsuccessful, retry until timeout exceeded.
    - Allowed: pinger <-> pinger
    - Allowed: real <-> real
    - Blocked: real <-> pinger
  - Add NetPol and assert connectivity. If unsuccessful, retry until timeout exceeded.
    - Allowed: real <-> pinger

#### 3. General CRUD at Medium Scale  (~30 fake Pods, 10 real Pods)
- Delete previous namespaces (including Pods and Netpols).
- Recreate everything in step 1.
- Delete and re-add labels.
- Delete and re-create Pods.
- Delete and re-create NetPols.

#### 4. Test Connectivity After CRUD
Must achieve initial connectivity within 10 minutes, and apply a new NetworkPolicy within 20 seconds.

- Check connectivity like in step 2.
